taskweaver/memory/shared_memory_entry.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Literal, Optional, Tuple
+
+from taskweaver.memory.type_vars import RoleName
+from taskweaver.utils import create_id
+
+
+@dataclass
+class SharedMemoryEntry:
+    type: Literal["plan", "experience_sub_path"]
+    by: str
+    content: str
+    scope: Literal["round", "conversation"]
+    # The id of the round or conversation
+    scope_id: str
```

Comment:
Does it necessary to specify the scope id when the id should be determined by where the shared memory is stored at?

Reply:
Yes, will try to remove it.